mmv1/relative/file_ops.go

```diff
@@ -0,0 +1,22 @@
+package relative
+
+import (
+	"os"
+	"path/filepath"
+)
+
+// For file IO relative to some base directory
+// Templates and other files are accessed
+// relative to os CWD, this library allows accessing a
+// location relative to an initialized global base
+
+var baseDirectory string
```

Comment:
We should avoid having globals like this that modify the behavior of a package. They tend to be difficult to manage and can lead to unexpected side effects. Instead, we should rely on a Loader instance to track its own base directory.

Reply:
This is impossible to set up through the loader. The YAML unmarshal function implemented in the examples file always reads the templates relative to repo root. Since we can't pass through the base location we need some sort of global control that.

Reply:
yeah, the fact that we need this information as part of ExecuteTemplate is an indication of a problem in the structure of the codebase.

One option that would be fairly simple which comes to mind is that the loader could convert relative links at load time (so that at generate time, that is, during ExecuteTemplate, the path is already resolved)

Reply:
Lets chat about this out of band I think

Reply:
I think this is reasonable. I can try

Reply:
changed to load examples by loader